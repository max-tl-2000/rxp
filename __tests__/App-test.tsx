import * as React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer, { act, ReactTestRenderer } from 'react-test-renderer';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
  SplashScreen: {
    preventAutoHide: jest.fn(),
  },
  Linking: {
    parse: () => '',
    getInitialURL: () => '',
  },
}));

jest.mock('../navigation/useLinking', () => ({ useLinking: () => ({ getInitialState: jest.fn() }) }));

jest.mock('../helpers/console-trap', () => ({}));

jest.mock('../components/AppBranding', () => {
  const { View } = require('react-native'); // eslint-disable-line
  return {
    AppBranding() {
      return <View />;
    },
  };
});

const App = require('../App').default; // eslint-disable-line

describe('App', () => {
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
    jest.useFakeTimers();
  });

  it('renders the root', async () => {
    let resolve: (value: ReactTestRenderer) => void;

    const renderPromise: Promise<ReactTestRenderer> = new Promise(r => {
      resolve = r;
    });

    await act(() => resolve(renderer.create(<App />)));

    const rdr = await renderPromise;
    expect(rdr.toJSON()).toMatchSnapshot();
  });
});
