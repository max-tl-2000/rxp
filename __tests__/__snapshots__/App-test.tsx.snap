// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the root 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  />
</View>
`;
